src/formatter-settings/index.ts

```diff
@@ -1,99 +1,356 @@
+                            await config.update(VSCodeSettings.INSERT_SPACES, targetValue, undefined, true);
+                        } else if (e.id === SupportedSettings.TABULATION_SIZE) {
+                            await config.update(VSCodeSettings.TAB_SIZE, Number(settingValue), undefined, true);
                         }
+                        this.profileSettings.set(e.id, settingValue);
+                    } else if (e.id === VSCodeSettings.DETECT_INDENTATION) {
+                        const config = vscode.workspace.getConfiguration(undefined, { languageId: "java" });
+                        await config.update(VSCodeSettings.DETECT_INDENTATION, (settingValue === "true"), undefined, true);
+                    } else {
+                        await this.modifyProfile(e.id, settingValue, document);
                     }
-                    webviewPanel.webview.postMessage({
-                        command: "loadProfileSetting",
-                        setting: Array.from(this.supportedProfileSettings.values()),
-                    });
-                    webviewPanel.webview.postMessage({
-                        command: "loadVSCodeSetting",
-                        setting: Array.from(this.supportedVSCodeSettings.values()),
-                    });
                     break;
                 default:
                     break;
             }
         });
+        await this.checkRequirement();
+    }
+
+    private async checkRequirement(): Promise<boolean> {
+        if (this.checkedRequirement) {
+            return true;
+        }
+        const javaExt = vscode.extensions.getExtension("redhat.java");
+        if (!javaExt) {
+            vscode.window.showErrorMessage("The extension 'redhat.java' is not installed. Please install it and run this command again.");
+            const err: Error = new Error("The extension 'redhat.java' is not installed.");
+            setUserError(err);
+            sendError(err);
+            return false;
+        }
+        const javaExtVersion: string = javaExt.packageJSON.version;
+        if (compareVersions(javaExtVersion, JavaConstants.MINIMUM_JAVA_EXTENSION_VERSION) < 0) {
+            vscode.window.showErrorMessage(`The extension version of 'redhat.java' is too stale. Please install at least ${JavaConstants.MINIMUM_JAVA_EXTENSION_VERSION} and run this command again.`);
+            const err: Error = new Error(`The extension version of 'redhat.java' (${javaExtVersion}) is too stale.`);
+            setUserError(err);
+            sendError(err);
+            return false;
+        }
+        await javaExt.activate();
+        this.checkedRequirement = true;
+        return true;
+    }
+
+    private async initialize(document: vscode.TextDocument, webviewPanel: vscode.WebviewPanel): Promise<boolean> {
+        if (!await this.checkRequirement()) {
+            return false;
+        }
+        if (!await this.checkProfile()) {
```

Comment:
why checkProfile() is called again here?

Reply:
The other `checkProfile()` is called in `showFormatterSettingsEditor()` when we execute command `java.formatterSettings`. If we open the editor in another way, such as `reload webviews` or manually open a xml using `open with...`, we need to check the profile.